internal/services/containers/kubernetes_cluster_resource_test.go

```diff
@@ -75,6 +75,20 @@ func TestAccKubernetesCluster_runCommand(t *testing.T) {
 	})
 }
 
+func TestAccKubernetesCluster_keyVaultKms(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_cluster", "test")
+	r := KubernetesClusterResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.azureKeyVaultKms(data, currentKubernetesVersion),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
```

Comment:
How about adding two more configs here, one for `enabled = false` and one for removing this new `key_vault_kms` block to prove that we could turn this feature off?

Reply:
I think same like https://github.com/hashicorp/terraform-provider-azurerm/pull/19893#discussion_r1064176562 - we don't actually create clusters. Therefore we cannot test what happens when we disable the feature on a KMS encrypted cluster. But in my manual tests it worked fine.

Reply:
Added case `enabled = false`. Not sure if we should add one with no `key_vault_kms {}` block since its already tested in every other test case. WDYT @lonegunmanb ?

Reply:
I would prefer an extra test case for no `key_vault_kms` block and the end to verify whether the code could be idempotent after user delete this block (sometimes it's not). But since this pull request should be reviewed by HashiCorp, I think we could let them call the shots.

Reply:
Not applicable anymore. I refactored to the recommendation from @stephybun and removed the enabled option.